MetaMask/metamask-mobile
# [Flask] new flask iOS target

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
- so that we can have a flask variant of the mobile app
- This PR **DOES NOT** setup the proper code signing or the proper bitrise pipelines.
_2. What is the improvement/solution?_
- defined a new target, I duplicated the MetaMask target so it should have almost all of the same settings
- added a ` yarn start:ios:flask` command to the `package.json`
    - This command calls the `build.sh` file with the arguments `/scripts/build.sh ios flaskDebug"`
- Modified the `build.sh` file to handle the `flaskDebug` variant which builds the app with the scheme set to `MetaMask-Flask`
- Added the flask logo with all the different sizes. The logo can be found [here](https://www.figma.com/file/f4i6WJ1rbxtD5TUk7mypnF/Mobile-Flask-Flows?type=design&node-id=2-675&t=2YnRuCLxVHrRpoNs-0).

The new bundle id will be `io.metamask.MetaMask-Flask`
<img width="462" alt="Screenshot 2023-06-14 at 10 48 28 PM" src="https://github.com/MetaMask/metamask-mobile/assets/22918444/ec4e52db-78bd-4831-8cfb-97489900a660">


**Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/22918444/5c5781b7-ef6e-475a-997b-dfe7c528b981


<img width="1057" alt="Screenshot 2023-06-14 at 10 45 22 PM" src="https://github.com/MetaMask/metamask-mobile/assets/22918444/5f2f8272-b841-4575-911c-fa8338df32b6">

### Testing/Reviewing
- checkout this branch and run `yarn setup` there should not be any major changes
- open xcode and in the top section beside the `play button`, click the MetaMask icon and a dropdown should open.
- then you should be able to select the `MetaMask-Flask` with the purple logo and click play

<img width="534" alt="Screenshot 2023-06-14 at 10 36 23 PM" src="https://github.com/MetaMask/metamask-mobile/assets/22918444/e5a94cbb-92b8-4287-aa54-d3190a985562">

- the app should build and you should be able to use it as normal but the logo should be different. You should be able to build the normal MetaMask flask target alongside Flask without updating. They should be separate. 
- To test the build.sh script,  run `yarn start:ios:flask`

<img width="851" alt="Screenshot 2023-06-14 at 10 42 46 PM" src="https://github.com/MetaMask/metamask-mobile/assets/22918444/7307dd00-bc35-4d32-90ac-ac438efa8d59">

- the build should kick off, it takes some time but after a minute or two it should be successful

<img width="540" alt="Screenshot 2023-06-14 at 10 41 52 PM" src="https://github.com/MetaMask/metamask-mobile/assets/22918444/624b6990-163d-4731-942d-84976bbcc128">


-  **NOTE**  The app will not launch in the simulator. I followed the same pattern as MetaMask-QA and noticed that that script builds but does not launch the simulator either so I think that this is alright...at least for now.
  


**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/1033

**Checklist**
